src/squlearn/util/executor.py

```diff
@@ -933,7 +937,7 @@ def get_shots(self) -> int:
 
             shots = max(shots_estimator, shots_sampler)
         elif self._backend is not None:
-            if "statevector_simulator" not in str(self._backend):
+            if "statevector" not in self._backend.configuration().backend_name:
```

Comment:
Did you try this with real backends? 
Sometimes backend.configuration() does not exist

Reply:
Works with each of the freely available backends, didn't check for paid backends

Reply:
If works for the free ones, its fine. Unfortunatly this change occurs in a couple of other places as well :(

Reply:
Sorry, see my review for an failing example :D